test/unit/minidump-loader.test.ts

```diff
@@ -0,0 +1,157 @@
+import { uuid4 } from '@sentry/utils';
+import { expect } from 'chai';
+import { appendFileSync, existsSync, utimesSync, writeFileSync } from 'fs';
+import { join } from 'path';
+import * as tmp from 'tmp';
+
+import { createMinidumpLoader } from '../../src/main/integrations/sentry-minidump/minidump-loader';
+
+function dumpFileName(): string {
+  return `${uuid4()}.dmp`;
+}
+
+const VALID_LOOKING_MINIDUMP = Buffer.from(`MDMP${'X'.repeat(12_000)}`);
+const LOOKS_NOTHING_LIKE_A_MINIDUMP = Buffer.from('X'.repeat(12_000));
+const MINIDUMP_HEADER_BUT_TOO_SMALL = Buffer.from('MDMPdflahfalfhalkfnaklsfnalfkn');
+
+describe('createMinidumpLoader', () => {
+  let tempDir: tmp.DirResult;
+  before(() => {
+    tempDir = tmp.dirSync({ unsafeCleanup: true });
+  });
+
+  after(() => {
+    if (tempDir) {
+      tempDir.removeCallback();
+    }
+  });
+
+  it('creates attachment from minidump', (done) => {
+    const name = dumpFileName();
+    const dumpPath = join(tempDir.name, name);
+    writeFileSync(dumpPath, VALID_LOOKING_MINIDUMP);
+
+    const loader = createMinidumpLoader(() => Promise.resolve([dumpPath]));
+
+    void loader(false, (attachment) => {
+      expect(attachment).to.eql({
+        data: VALID_LOOKING_MINIDUMP,
+        filename: name,
+        attachmentType: 'event.minidump',
+      });
+
+      setTimeout(() => {
+        expect(existsSync(dumpPath)).to.be.false;
+        done();
+      }, 1_000);
+    });
+  });
+
+  it("doesn't send invalid minidumps", (done) => {
+    const missingHeaderDump = join(tempDir.name, dumpFileName());
+    writeFileSync(missingHeaderDump, LOOKS_NOTHING_LIKE_A_MINIDUMP);
+    const tooSmallDump = join(tempDir.name, dumpFileName());
+    writeFileSync(tooSmallDump, MINIDUMP_HEADER_BUT_TOO_SMALL);
+
+    const loader = createMinidumpLoader(() => Promise.resolve([missingHeaderDump, tooSmallDump]));
+
+    let passedAttachment = false;
+    void loader(false, () => {
+      passedAttachment = true;
+    });
+
+    setTimeout(() => {
+      expect(passedAttachment).to.be.false;
+      expect(existsSync(missingHeaderDump)).to.be.false;
+      expect(existsSync(tooSmallDump)).to.be.false;
+      done();
+    }, 2_000);
+  });
+
+  it("doesn't send minidumps that are over 30 days old", (done) => {
```

Comment:
This test fails in CI on Windows and Linux but passes on macOS in CI and locally 🤷‍♂️

Reply:
yeah this is weird - so `utimesSync` doesn't work? 

Can we change file location for test?

Reply:
Just tested Windows locally and that's not working either. Looking into it. 

Tried `Math.floor(new Date().getTime() / 1000)` in case it was the decimal that was upsetting it!